simulation_ws/src/test_nodes/nodes/docking_test.py

```diff
@@ -0,0 +1,152 @@
+#!/usr/bin/env python
+
+import rospy
+import rostest
+import time
+import os
+import unittest
+import tf
+
+from math import hypot
+from std_msgs.msg import Bool
+from geometry_msgs.msg import PoseStamped
+from rosgraph_msgs.msg import Clock
+from nav_msgs.msg import Odometry
+from std_srvs.srv import Trigger
+
+from robomaker_simulation_msgs.msg import Tag
+from robomaker_simulation_msgs.srv import Cancel, AddTags
+
+class DockingTest(unittest.TestCase):
+
+    def runTest(self):
+        # Start the docking test
+        self.test_docking()
+
+    def setUp(self):
+        self.test_name = "Robot_Monitoring_Tests_" + \
+            str(time.time()).split(".", 1)[0]
+        rospy.loginfo("Test Name: %s", self.test_name)
+        self.is_docked = False
+        self.is_docking = False
+        self.get_params()
+        while rospy.Time.now().secs == 0:
+            pass
+        self.starting_time_secs = rospy.Time.now().secs
+        self.timer = rospy.Timer(30, callback=self.timer_cb(), oneshot=True)
```

Comment:
`timer_cb` does not tell me what it will do.
I'd suggest the callback to be `trigger_docking_client` as it is, just adding the `self.is_docking = True`
The Timer duration should be a parameter, it is how long the robot should be moving before docking